csharp/ql/test/query-tests/Security Features/CWE-798/HardcodedCredentials.expected

```diff
@@ -1,23 +1,42 @@
 edges
+| HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" |
+| HardcodedCredentials.cs:33:19:33:28 | "username" | HardcodedCredentials.cs:33:19:33:28 | "username" |
+| HardcodedCredentials.cs:47:39:47:53 | "myNewPa55word" | HardcodedCredentials.cs:47:39:47:53 | "myNewPa55word" |
 | HardcodedCredentials.cs:49:30:49:60 | array creation of type Byte[] | HardcodedCredentials.cs:52:13:52:23 | access to local variable rawCertData |
-nodes
-| HardcodedCredentials.cs:17:25:17:36 | "myPa55word" |
-| HardcodedCredentials.cs:33:19:33:28 | "username" |
-| HardcodedCredentials.cs:47:39:47:53 | "myNewPa55word" |
-| HardcodedCredentials.cs:49:30:49:60 | array creation of type Byte[] |
-| HardcodedCredentials.cs:52:13:52:23 | access to local variable rawCertData |
-| HardcodedCredentials.cs:53:13:53:24 | "myPa55word" |
-| HardcodedCredentials.cs:76:31:76:42 | "myusername" |
-| HardcodedCredentials.cs:76:45:76:56 | "mypassword" |
-| TestHardcodedCredentials.cs:21:31:21:42 | "myusername" |
-| TestHardcodedCredentials.cs:21:45:21:56 | "mypassword" |
-| TestHardcodedCredentials.cs:26:19:26:28 | "username" |
+| HardcodedCredentials.cs:53:13:53:24 | "myPa55word" | HardcodedCredentials.cs:53:13:53:24 | "myPa55word" |
+| HardcodedCredentials.cs:76:31:76:42 | "myusername" | HardcodedCredentials.cs:76:31:76:42 | "myusername" |
+| HardcodedCredentials.cs:76:45:76:56 | "mypassword" | HardcodedCredentials.cs:76:45:76:56 | "mypassword" |
+| TestHardcodedCredentials.cs:21:31:21:42 | "myusername" | TestHardcodedCredentials.cs:21:31:21:42 | "myusername" |
+| TestHardcodedCredentials.cs:21:45:21:56 | "mypassword" | TestHardcodedCredentials.cs:21:45:21:56 | "mypassword" |
+| TestHardcodedCredentials.cs:26:19:26:28 | "username" | TestHardcodedCredentials.cs:26:19:26:28 | "username" |
 #select
 | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | The hard-coded value "myPa55word" flows to $@ which is compared against $@. | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | "myPa55word" | HardcodedCredentials.cs:17:13:17:20 | access to local variable password | access to local variable password |
+| HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | The hard-coded value "myPa55word" flows to $@ which is compared against $@. | HardcodedCredentials.cs:17:25:17:36 | "myPa55word" | "myPa55word" | HardcodedCredentials.cs:17:13:17:20 | access to local variable password | access to local variable password |
```

Comment:
There seems to be some unnecessary duplication of the results. I think these duplicated results must be removed (also in other queries).

Reply:
Good catch, this is actually because of a bug in the query; it uses `hasFlow`, but should have used `hasFlowPath`.